keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1800 @@
+    represents the set of resources within the pod's total NUMA-aligned budget
+    that are available to be shared by all containers that do not have their own
+    exclusive `Guaranteed` resource requests. This mechanism is key to the
+    partitioning model, as it ensures that critical containers receive their
+    reserved resources while allowing all other containers to efficiently share
+    the remainder of the pod's budget.
+
+3.  **Handling Init Containers:**
+
+    -   **Standard Init Containers:** For init containers that are individually
+        `Guaranteed`, an exclusive slice is carved out from the pool. Upon
+        completion, these CPUs are added to a per-pod `reusable` pool, making
+        them available to subsequent app containers within the pool.
+    -   **Restartable Init Containers (Sidecars):** For restartable init
+        containers that are individually `Guaranteed`, an exclusive slice is
+        carved out and reserved for the entire pod lifecycle. These CPUs are
+        *not* added to the reusable pool.
+
+4.  **[New Logic] Pre-computation and State Update:** The manager pre-computes
+    and writes the specific assignment for every container to the state file.
+    `Guaranteed` containers are assigned their exclusive slices, and all other
+    containers are assigned the `podSharedPool`.
+
+5.  **Container-Level Allocation:** For the current container (and all
+    subsequent containers in the pod), the `Allocate` function reads that
+    container's specific, pre-computed assignment from the state and applies it
+    to the container's cgroup.
+
+#### State Management and Container Removal
+
+The in-memory state and on-disk checkpoint file will be updated to be pod-aware.
+For the `pod` scope, the state will now store the total managed resource set for
+the pod and the individual assignments for each container (either an exclusive
+slice or the `podSharedPool`).
+
+The `RemoveContainer` function will also be updated. When a container with an
+exclusive slice is removed, its resources will be returned to the pod's shared
+pool. The pod's entire managed resource set will only be released back to the
+node's shared pool when the last container of that pod is removed.
+
+### Memory Manager
+
+The Memory Manager will be updated to support pod-level resources when its
+policy is set to `Static`. The core changes will affect how it determines the
+scope of resource management and how it allocates memory to containers within
+that scope.
+
+-   **Pod Scope:** When the Topology Manager's scope is `pod`, the Memory
+    Manager will manage a set of memory blocks on a single NUMA node for the
+    entire pod, based on `pod.spec.resources`. This memory will then be
+    partitioned, containers with `Guaranteed` memory requests will be allocated
+    exclusive memory blocks from this set, while the remaining memory will form
+    a shared pool to be allocated to all other containers in the pod.
```

Comment:
While I'm personally onboard because this seems the right move, we're effectively changing the semantic of the existing API because now the same setting do a different thing. Let's doublecheck this is fine to do. This will probably be covered by PRR review, but still let's make sure this is covered explicitly (this concern applies to all the other similar instances ofc)